communicators/http_communicator.go

```diff
@@ -47,37 +48,82 @@ func (m *HTTPCommunicator) RespondWith(responses []prototype.MessagePrototype) {
 			resp := model.ResponseMessage{}
 			for r := range res.Mediators.Run(message, resp) {
 				if matchers.MatchAny(res.Matcher, message) {
-					io.WriteString(w, r.Response)
+					io.WriteString(w, r.Body)
 				}
 			}
 
 		}
 	}
 }
 
+// RequestWith sets the request handler
+func (m *HTTPCommunicator) RequestWith(requests []prototype.MessagePrototype) {
+	m.requests = requests
+}
+
 // Close closes connection
 func (m *HTTPCommunicator) Close() error {
 	panic("not implemented") // TODO: Implement
 }
 
 // Post posts requests to endpoint
 func (m *HTTPCommunicator) Post(body model.RequestMessage) error {
-	_, err := m.sendRequest(string(body.Body))
+	_, err := m.sendRequest("", "GET", string(body.Body))
 
 	return err
 }
 
+// PostAndListen posts requests to endpoint and listens for response
+func (m *HTTPCommunicator) PostAndListen() (chan model.ResponseMessage, error) {
+	responseChan := make(chan model.ResponseMessage)
+	signalChannel := make(chan struct{})
+
+	for _, request := range m.requests {
+		go func(request prototype.MessagePrototype, sigChannel chan struct{}) {
+			resp := model.ResponseMessage{}
+			for r := range request.Mediators.Run(model.RequestMessage{}, resp) {
+				resp, err := m.sendRequest(request.To, request.Type, r.Body)
+				if err != nil {
+					log.Printf("Error sending request: %s", err)
+				}
+
+				result := model.ResponseMessage{
+					Body: readResponseBody(resp),
+					Metadata: map[string]string{
+						"status": resp.Status,
```

Comment:
Move these metadata names to some const. In tests, it's better to use direct names to indicate some changes yet for code it is better to have some const